baseplate/experiments/providers/simple_experiment.py

```diff
@@ -0,0 +1,150 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import logging
+import hashlib
+import time
+
+from .base import Experiment
+from ..._compat import long, iteritems
+
+
+logger = logging.getLogger(__name__)
+
+
+class SimpleExperiment(Experiment):
+    """Base class for simple experiments. Note that this class shares a lot of
+    code from the r2.py class, which is slated for deprecation and removal.
+    """
+
+    def make_seed(self, seed_data):
+        id = seed_data.get('id')
+        name = seed_data.get('name')
+        version = seed_data.get('shuffle_version')
+
+        return "{}.{}.{}".format(id, name, version)
+
+    def __init__(self, id, name, owner, start_ts, stop_ts, config,
+                 enabled=True, **kwargs):
+
+        self.id = id
+        self.name = name
+        self.owner = owner
+        self.num_buckets = 1000
+
+        self.start_ts = start_ts
+        self.stop_ts = stop_ts
+        self.enabled = enabled
+
+        self.bucket_val = config.get("bucket_val", "user_id")
+        self.version = config.get("experiment_version")
+        self.shuffle_version = config.get("shuffle_version")
+
+        self.variants = config.get("variants", [])
+
+        seed_data = {"id": id, "name": name, "shuffle_version": self.shuffle_version}
+        self.seed = config.get("bucket_seed", self.make_seed(seed_data))
```

Comment:
I think the general preference is to do this parsing in the factory methods (i.e. `from_dict`) and pass everything into the constructor explicitly rather than having it done in the constructor.

Reply:
Is that a python thing, a Reddit thing, or a "this is how it was done in here previously, so please stay consistent" thing?  I realize that currently my `from_dict` method is pretty useless, but I'm just trying to determine the reasoning for needing that method at all is.

Reply:
It's more of 3 thing, although that's not always consistent.  You can see other examples of this in other parts of baseplate such as https://github.com/reddit/baseplate/blob/69eba73ad7493e8c4822677b391346b5a8ee7c38/baseplate/queue_consumer.py#L113

Reply:
IMO, my reason for preferring this is I like the constructor to make it explicit what the dependencies of that object are.  When you can just pass in a couple of things and build off of that, you have to dive into the constructor to try to figure that out.